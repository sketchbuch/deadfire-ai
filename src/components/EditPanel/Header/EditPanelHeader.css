/* Edit Panel Header */
/* ----------------- */

.EditPanelHeader {
  position: absolute;
  display: flex;
  flex-direction: row;
  flex-wrap: no-wrap;
  justify-content: flex-start;
  align-items: center;
  top: 0px;
  left: 0px;
  margin: 0px;
  padding: 0px 20px;
  width: 100%;
  height: var(--sidebar-header-height);
  background: var(--col-lighten-1);
  border-bottom: 1px solid var(--col-darken-1);
  z-index: var(--z-puller);
}

.EditPanelHeader::after {
  opacity: 0.1;
  content: ' ';
  position: absolute;
  bottom: -11px;
  left: 0px;
  width: 100%;
  height: 10px;
  display: block;

  /* http://colorzilla.com/gradient-editor/#000000+0,000000+100&1+0,0+100 */
  background: -moz-linear-gradient(top, rgba(0, 0, 0, 1) 0%, rgba(0, 0, 0, 0) 100%);
  background: -webkit-linear-gradient(top, rgba(0, 0, 0, 1) 0%, rgba(0, 0, 0, 0) 100%);
  background: linear-gradient(to bottom, rgba(0, 0, 0, 1) 0%, rgba(0, 0, 0, 0) 100%);
  filter: progid:DXImageTransform.Microsoft.gradient( startColorstr='#000000', endColorstr='#00000000',GradientType=0 );
}

.EditPanelHeader__headline {
  margin: 0px;
  font-size: var(--font-size);
  font-weight: normal;
  overflow: hidden;
  display: block;
  text-overflow: ellipsis;
  white-space: nowrap;
  color: var(--col-white);
  width: 50%;
}

.EditPanelHeader__headline span[data-count] {
  display: inline-block;
  background: transparent;
  color: var(--col-lighten-5);
  margin-left: 10px;
  padding: 0px 6px;
}

.EditPanelHeader__headline span[data-alert] {
  display: inline-block;
  background: transparent;
  color: var(--col-lighten-5);
  margin-left: 10px;
  padding: 0px 6px;
  border-radius: var(--border-radius);
}

.EditPanelHeader__headline span[data-alert='true'] {
  background: var(--col-neg-dark);
  color: var(--col-lighten-7);
  border: 1px solid var(--col-darken-2);
}

.EditPanelHeader__headline span:empty {
  display: none;
  padding: 0px;
  margin: 0px;
}

.EditPanelHeader__controls {
  padding-left: 20px;
  position: relative;
}

.EditPanelHeader__search {
  background: var(--col-darken-1);
  border: 1px solid var(--col-darken-2);
  height: 32px;
  color: var(--col-white);
  padding-right: 40px;
}

.EditPanelHeader__search::-webkit-input-placeholder {
  color: var(--col-lighten-5);
}

.EditPanelHeader__searchclear {
  width: 32px;
  height: 32px;
  display: none;
  position: absolute;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  right: 0px;
  top: 0px;
  font-size: var(--font-size-medium);
  cursor: pointer;
  color: var(--col-white);
}

.EditPanelHeader__search:not(:placeholder-shown) + .EditPanelHeader__searchclear {
  display: flex;
}
