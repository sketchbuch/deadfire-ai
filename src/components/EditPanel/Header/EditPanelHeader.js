// @flow

import * as React from 'react';
import './EditPanelHeader.css';

type Props = {
  children?: React.Node,
  title: string,
};

/**
 * Edit Panel header.
 */
export class EditPanelHeader extends React.Component<Props> {
  static defaultProps = {
    children: null,
    title: '',
  };

  props: Props;

  render() {
    return (
      <header className="EditPanelHeader">
        {this.props.children ? this.props.children : <h1 className="EditPanelHeader__headline">{this.props.title}</h1>}
      </header>
    );
  }
}

export default EditPanelHeader;
