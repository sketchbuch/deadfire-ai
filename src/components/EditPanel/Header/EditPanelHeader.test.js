// @flow

import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import EditPanelHeader from './EditPanelHeader';

describe('<EditPanelHeader />', () => {
  const props = { title: 'Test title' };

  test('Renders without crashing', () => {
    const wrapper = shallow(<EditPanelHeader {...props} />);
    expect(wrapper).toHaveLength(1);
  });

  test('Props render correctly', () => {
    const testRender = renderer.create(<EditPanelHeader {...props} />).toJSON();
    expect(testRender).toMatchSnapshot();
  });
});
