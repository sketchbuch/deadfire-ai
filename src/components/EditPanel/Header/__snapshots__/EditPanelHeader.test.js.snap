// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EditPanelHeader /> Props render correctly 1`] = `
<header
  className="EditPanelHeader"
>
  <h1
    className="EditPanelHeader__headline"
  >
    Test title
  </h1>
</header>
`;
